app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationsViewModel.kt

```diff
@@ -500,6 +516,41 @@ class NotificationsViewModel @Inject constructor(
             }
     }
 
+    fun hasNewestNotificationId(type: Notification.Type, statusId: String, notificationId: String): Boolean {
+        val trackerArray = when(type) {
+            Notification.Type.FAVOURITE -> seenFavorites
+            Notification.Type.REBLOG -> seenBoosts
+            else -> null
+        } ?: return true
+
+        val highestNotificationId = trackerArray[statusId]
+
+        return if (highestNotificationId == null || isEqualOrNewer(notificationId, highestNotificationId)) {
+            trackerArray[statusId] = notificationId
+
+            true
+        } else {
+            // TODO edge case: a newer favorite has been removed: the old notification will not be added again
```

Comment:
I wonder if this edge case is worth a fix.

(It will be correct for an app restart.)

Reply:
Have been trying to do the heuristic now. But I didn't really find a suitable place in all that paging and flowing.

(Added a few Todo comments, though.)